metaflow/plugins/pypi/parsers.py

```diff
@@ -0,0 +1,253 @@
+# this file can be overridden by extensions as is (e.g. metaflow-nflx-extensions)
+
+
+def requirements_txt_parser(content: str):
+    """
+    Parse non-comment lines from a requirements.txt file as strictly valid
+    PEP 508 requirements.
+
+    Recognizes direct references (e.g. "my_lib @ git+https://..."), extras
+    (e.g. "requests[security]"), and version specifiers (e.g. "==2.0"). If
+    the package name is "python", its specifier is stored in the "python"
+    key instead of "packages".
+
+    Parameters
+    ----------
+    content : str
+        Contents of a requirements.txt file.
+
+    Returns
+    -------
+    dict
+        A dictionary with two keys:
+            - "packages": dict(str -> str)
+              Mapping from package name (plus optional extras/references) to a
+              version specifier string.
+            - "python": str or None
+              The Python version constraints if present, otherwise None.
+
+    Raises
+    ------
+    ValueError
+        If a requirement line is invalid PEP 508 or if environment markers are
+        detected, or if multiple Python constraints are specified.
+    """
+    import re
+    from metaflow._vendor.packaging.requirements import Requirement, InvalidRequirement
+
+    parsed = {"packages": {}, "python": None}
+
+    inline_comment_pattern = re.compile(r"\s+#.*$")
+    for line in content.splitlines():
+        line = line.strip()
+
+        if not line or line.startswith("#"):
+            continue
+
+        line = inline_comment_pattern.sub("", line).strip()
+        if not line:
+            continue
+
```

Comment:
the previous PR had more specific line handling for requirements.txt content, specifically errors for options:
https://pip.pypa.io/en/stable/reference/requirements-file-format/#global-options

should we reintroduce these, or do we want to keep this implementation as simple as possible, and simply error out on 'unparseable' lines in the file? right now the parser throws an exception if the requirements.txt contains any options, be it global or per-requirement